bin_tests/tests/test_the_tests.rs

```diff
@@ -0,0 +1,48 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+#![cfg(unix)]
+
+use std::{fs, process};
+
+use bin_tests::{build_artifacts, ArtifactType, ArtifactsBuild, Profile};
+
+#[test]
+#[cfg_attr(miri, ignore)]
+fn test_the_tests_debug() {
+    test_the_tests_inner(Profile::Debug);
+}
+
+#[test]
+#[cfg_attr(miri, ignore)]
+fn test_the_tests_release() {
+    test_the_tests_inner(Profile::Release);
+}
+
+fn test_the_tests_inner(profile: Profile) {
```

Comment:
It took me a ⏱️ to realize that `Profile` is not the same as `Profile` in profiling.  Maybe use a different name?

Reply:
Changed to `BuildProfile`